docs-js/tutorials/getting-started-with-agents.mdx

```diff
@@ -29,199 +29,108 @@ The assistant performs these steps:
 
 ## Prerequisites
 
-Refer to prerequisites outlined [here](../../overview-cloud-sdk-for-ai-js#prerequisites).
+Refer to the prerequisites outlined [here](../../overview-cloud-sdk-for-ai-js#prerequisites).
 
-This tutorial assumes you have a basic understanding of TypeScript, LLMs and LangChain concepts.
+This tutorial assumes a basic understanding of TypeScript, LLMs, and LangChain concepts.
 
 ## Installation
 
 Install the following dependencies:
 
 ```bash
-npm install @sap-ai-sdk/langchain langchain @langchain/langgraph @langchain/core zod
+npm install @sap-ai-sdk/langchain langchain @langchain/langgraph @langchain/core @langchain/mcp-adapters @modelcontextprotocol/sdk zod
 ```
 
-## Define Mock Tools
+## Define Tools
 
-Define tools:
+The following tools are defined:
 
-- `get_weather` for returning weather by city
-- `get_restaurants` for city-based restaurant recommendations
+- `get_weather` for returning weather by city.
+- `get_restaurants` for city-based restaurant recommendations.
 
 :::info
-This example uses mocked data for simplicity.
-Replace the tools' implementation with real API calls to integrate live data.
+The `get_restaurants` tool uses mocked data for simplicity.
+Replace with real API calls to integrate live data.
 :::
 
-```ts
-import { tool } from '@langchain/core/tools';
-import { z } from 'zod';
-
-const mockWeatherData: Record<
-  string,
-  { temperature: string; condition: string }
-> = {
-  paris: { temperature: '22°C', condition: 'Mild and pleasant' },
-  reykjavik: { temperature: '3°C', condition: 'Cold and windy' }
-};
-
-const mockRestaurantData: Record<string, string[]> = {
-  paris: ['Le Comptoir du Relais', "L'As du Fallafel", 'Breizh Café'],
-  reykjavik: ['Dill Restaurant', 'Fish Market', 'Grillmarkaðurinn']
-};
-
-const getWeatherTool = tool(
-  async ({ city }) => {
-    const weather = mockWeatherData[city.toLowerCase()];
-    return weather
-      ? `Current weather in ${city}: ${weather.temperature}, ${weather.condition}`
-      : `Weather data not available for ${city}.`;
-  },
-  {
-    name: 'get_weather',
-    description: 'Get current weather information for a city',
-    schema: z.object({ city: z.string().describe('The city name') })
-  }
-);
-
-const getRestaurantsTool = tool(
-  async ({ city }) => {
-    const restaurants = mockRestaurantData[city.toLowerCase()];
-    return restaurants
-      ? `Popular restaurants in ${city}: ${restaurants.join(', ')}`
-      : `No restaurant data available for ${city}.`;
-  },
-  {
-    name: 'get_restaurants',
-    description: 'Get restaurant recommendations for a city',
-    schema: z.object({ city: z.string().describe('The city name') })
-  }
-);
+```js reference referenceLinkText=" " title="Weather tool definition"
+https://github.com/SAP/ai-sdk-js/blob/main/sample-code/src/tutorials/travel-itinerary-agent.ts#L23-L42
```

Comment:
The only problem I can see here is that, if we update the sample-code, we need to check everytime if it is used in our docs portal and make sure to adapt the line numbers in docs as well.

Maybe we can nail this to some commit hash? Or add a comment in the sample code to indicate that this block will affect docs?